lmdeploy/pytorch/config.py

```diff
@@ -218,11 +228,15 @@ def __build_qwen():
             model_config = __build_dbrx()
         elif hf_config.model_type == 'qwen':
             model_config = __build_qwen()
+        elif model_arch == 'CogVLMForCausalLM':
+            model_config = __build_cogvlm()
         else:
             model_config = __build_default()
 
         model_config.dtype = _get_torch_dtype(hf_config)
 
         model_config.hf_config = hf_config
-        model_config.json_config = hf_config.to_dict()
+        model_config.json_config = json_config
+        model_config.model_arch = model_arch
```

Comment:
model_arch will not be used in other modules, these is no need to save it in config.